cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -162,3 +168,327 @@ def zones(self):
                  if zone['region'] == self._gce_region['selfLink']]
         return [GCEPlacementZone(self._provider, zone['name'], self.name)
                 for zone in zones]
+
+
+class GCEFirewallsDelegate(object):
+
+    _instance = None
+   
+    def __init__(self, provider):
+        if GCEFirewallsDelegate._instance is not None:
+            raise Exception('Use GCEFirewalls.instance() to instanciate.')
+        self._provider = provider
+        self._list_response = None
+        GCEFirewallsDelegate._instance = self
```

Comment:
It looks like this class is meant to be a singleton? Will there be issues if there are multiple provider instances, each having different connection settings, and they call GCEFirewallsDelegate.get_instance() alternately, because it's caching the provider within the singleton?